src/Mvc/Mvc.ApiExplorer/src/ApiResponseTypeProvider.cs

```diff
@@ -214,15 +214,16 @@ private Type GetDeclaredReturnType(ControllerActionDescriptor action)
         {
             var declaredReturnType = action.MethodInfo.ReturnType;
             if (declaredReturnType == typeof(void) ||
-                declaredReturnType == typeof(Task))
+                declaredReturnType == typeof(Task) ||
+                declaredReturnType == typeof(ValueTask))
             {
                 return typeof(void);
             }
 
             // Unwrap the type if it's a Task<T>. The Task (non-generic) case was already handled.
             var unwrappedType = declaredReturnType;
             if (declaredReturnType.IsGenericType &&
-                declaredReturnType.GetGenericTypeDefinition() == typeof(Task<>))
+                (declaredReturnType.GetGenericTypeDefinition() == typeof(Task<>) || declaredReturnType.GetGenericTypeDefinition() == typeof(ValueTask<>)))
```

Comment:
note that this is going to handle `ValueTask` and not other custom awaitables. given that we haven't gotten any feedback about custom awaitables, that seems fine.